flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -155,16 +158,29 @@ public Builder tableSchema(TableSchema newTableSchema) {
       return this;
     }
 
+    public Builder overwrite(boolean newOverwrite) {
+      this.overwrite = newOverwrite;
+      return this;
+    }
+
     @SuppressWarnings("unchecked")
     public DataStreamSink<RowData> build() {
       Preconditions.checkArgument(rowDataInput != null,
           "Please use forRowData() to initialize the input DataStream.");
-      Preconditions.checkNotNull(table, "Table shouldn't be null");
       Preconditions.checkNotNull(tableLoader, "Table loader shouldn't be null");
       Preconditions.checkNotNull(hadoopConf, "Hadoop configuration shouldn't be null");
 
+      if (table == null) {
+        tableLoader.open(hadoopConf);
+        try (TableLoader loader = tableLoader) {
+          this.table = loader.loadTable();
+        } catch (IOException e) {
+          throw new UncheckedIOException("Failed to load iceberg table.", e);
```

Comment:
Minor: it would be nice to have more context here. Maybe the table loader should define a `toString` that could be used in the error message here.

Reply:
Defining the `toString` sounds good to me.